WordPress/Classes/Utility/WPUserAgent.m

```diff
@@ -0,0 +1,88 @@
+#import "WPUserAgent.h"
+
+static NSString* const WPUserAgentKeyUserAgent = @"UserAgent";
+static NSString* const WPUserAgentKeyDefaultUserAgent = @"DefaultUserAgent";
+static NSString* const WPUserAgentKeyWordPressUserAgent = @"AppUserAgent";
+
+@implementation WPUserAgent
+
+#pragma mark - Init
+
+- (instancetype)init
+{
+    self = [super init];
+    
+    if (self) {
+        [self setupUserAgent];
+    }
+    
+    return self;
+}
+
+#pragma mark - One time setup
+
+- (void)setupUserAgent
+{
+    // Keep a copy of the original userAgent for use with certain webviews in the app.
+    NSString *defaultUA = [[[UIWebView alloc] init] stringByEvaluatingJavaScriptFromString:@"navigator.userAgent"];
+    NSString *wordPressUserAgent = [self wordPressUserAgent];
+
+    NSDictionary *dictionary = @{
+                                 WPUserAgentKeyUserAgent : wordPressUserAgent,
+                                 WPUserAgentKeyDefaultUserAgent : defaultUA,
+                                 WPUserAgentKeyWordPressUserAgent : wordPressUserAgent
+                                 };
+    [[NSUserDefaults standardUserDefaults] registerDefaults:dictionary];
+}
+
+#pragma mark - WordPress User-Agent
+
+- (NSString *)wordPressUserAgent
+{
+    UIDevice *device = [UIDevice currentDevice];
+    NSString *appVersion = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"];
+    NSString *userAgent = [NSString stringWithFormat:@"wp-iphone/%@ (%@ %@, %@) Mobile",
+                           appVersion,
+                           device.systemName,
+                           device.systemVersion,
+                           device.model];
+    
+    return userAgent;
+}
+
+#pragma mark - Changing the user agent
+
+- (void)useDefaultUserAgent
+{
+    NSString *ua = [[NSUserDefaults standardUserDefaults] stringForKey:WPUserAgentKeyDefaultUserAgent];
+    
+    [self setUserAgent:ua];
+}
+
+- (void)useWordPressUserAgent
+{
+    NSString *ua = [[NSUserDefaults standardUserDefaults] stringForKey:WPUserAgentKeyWordPressUserAgent];
+    
+    [self setUserAgent:ua];
+}
+
+- (void)setUserAgent:(NSString*)userAgent
+{
+    NSParameterAssert([userAgent isKindOfClass:[NSString class]]);
+    
+    NSDictionary *dictionary = [[NSDictionary alloc] initWithObjectsAndKeys:userAgent, WPUserAgentKeyUserAgent, nil];
```

Comment:
Why not use the object literal syntax? e.g. `NSDictionary *dictionary = @{ WPUserAgentKeyUserAgent: userAgent };`